opendevin/core/metrics.py

```diff
@@ -0,0 +1,54 @@
+class Metrics:
+    """
+    Metrics class can record various metrics during running and evaluation.
+    Currently we define the following metrics:
+        total_cost: the total cost of the current LLM.
+    """
+
+    def __init__(self) -> None:
+        self._total_cost: float = 0.0
+        self._costs: list[float] = []
+        self.reset()
+
+    @property
+    def total_cost(self) -> float:
+        return self._total_cost
+
+    @total_cost.setter
+    def total_cost(self, value: float) -> None:
+        if value < 0:
+            raise ValueError('Total cost cannot be negative.')
+        self._total_cost = value
+
+    @property
+    def costs(self) -> list:
+        return self._costs
+
+    def add_cost(self, value: float) -> None:
+        if value < 0:
+            raise ValueError('Added cost cannot be negative.')
+        self._total_cost += value
+        self._costs.append(value)
+
+    def reset(self):
```

Comment:
This method could be removed if it's not being used? I cannot think of when we want to reset it.